pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -50,3 +50,7 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, infrastructure *
 		SetEnvVars(generateTerraformInfraVariablesEnvironment(infrastructure.Spec.SecretRef)...).
 		Destroy(ctx)
 }
+
+func (a *actuator) ForceDelete(_ context.Context, _ logr.Logger, _ *extensionsv1alpha1.Infrastructure, _ *extensionscontroller.Cluster) error {
+	return nil
+}
```

Comment:
what is the expected behavior to actually support ForceDelete?
e.g. the gcp extension is waiting for the managed resource deletion https://github.com/gardener/gardener-extension-provider-gcp/pull/659/files#diff-c2e01c540745a0df87bb8d3b604303079a1f0b70aa33e17b649dd0448fa54f75

Should we check here for resources that might leak external infra?

Reply:
ping @shafeeqes

Reply:
> what is the expected behavior to actually support ForceDelete? e.g. the gcp extension is waiting for the managed resource deletion https://github.com/gardener/gardener-extension-provider-gcp/pull/659/files#diff-c2e01c540745a0df87bb8d3b604303079a1f0b70aa33e17b649dd0448fa54f75
> 
> Should we check here for resources that might leak external infra?

👀 
The gcp extension also returns nil for infrastructure delete.
Without valid credentials, we cannot clean them up, that's why we return nil.